### PR TITLE
Feat: add GitOps doc

### PR DESCRIPTION
**Pull request by FogDong**
add GitOps doc

/cc @wonderflow @BinaryHB0916 @hongchaodeng

### CODE REVIEW

File: docs/case-studies/gitops.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        http:
+          /: 9898
+  - name: nginx-server
+    type: webservice
+    properties:
+      image: nginx:1.21
+      port: 80
+```
+
+Second, apply a secret with your Git username and password in cluster:
```
Comment (zzxwill) [inline docs/case-studies/gitops.md]: Can I ignore the second step if the repo is public?

File: docs/case-studies/gitops.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  name: my-secret
+type: kubernetes.io/basic-auth
+stringData:
+  username: your username
+  password: your password
+```
+
+## Write the Application that can apply automatically
+
+Apply the `Application` file to the cluster:
```
Comment (zzxwill) [inline docs/case-studies/gitops.md]: We have two applications here, one is from the git repo above, and the one as below. Maybe specifying the `application file` a little bit more will help:)

File: docs/case-studies/gitops.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+      repoType: git
+      # your git repository url
+      url: https://github.com/FogDong/KubeVela-GitOps-Demo
+      # your secret name
+      secretRef: my-secret
+      # the interval time of pull config from repo
+      pullInterval: 1m
+      git:
+        branch: master
+      path: .
```
Comment (zzxwill) [inline docs/case-studies/gitops.md]: We normally run `vela show xxx` to guide users how to fill the properties.

BTW, my definition is out-dated?

```
➜  kubevela.io git:(cloud-resource) ✗ vela show kustomize
I0908 14:57:10.474122   91708 request.go:668] Waited for 1.014010181s due to client-side throttling, not priority and fairness, request: GET:https://47.242.209.62:6443/apis/image.toolkit.fluxcd.io/v1alpha1?timeout=32s
# Properties
+--------------+------------------------------------------------------------------------------------------------------+--------+----------+---------+
|     NAME     |                                             DESCRIPTION                                              |  TYPE  | REQUIRED | DEFAULT |
+--------------+------------------------------------------------------------------------------------------------------+--------+----------+---------+
| path         | Path to the directory containing the kustomization.yaml file, or the set of plain YAMLs a            | string | true     |         |
|              | kustomization.yaml should be generated for.                                                          |        |          |         |
| pullInterval | The interval at which to check for repository updates.                                               | string | true     | 1m      |
| repoUrl      | The repository URL, can be a HTTP/S or SSH address.                                                  | string | true     |         |
| branch       | The Git reference to checkout and monitor for changes, defaults to master branch.                    | string | true     | master  |
+--------------+------------------------------------------------------------------------------------------------------+--------+----------+---------+
```

File: docs/case-studies/gitops.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  workflow:
+    steps:
+      - name: multi-env
+        type: multi-env
+        properties:
+          policy: my-policy
+          env: my-env
+```
+
+Check the `Application` in clusters, we can see the application has been applied to `prod` namespace after a while:
```
Comment (zzxwill) [inline docs/case-studies/gitops.md]: What will happen to the application in `default` namespace?

**Comment by zzxwill** [comment]
This example combines the feature of GitOps and multiple-clusters deploying, which is awesome!

I wonder whether it's necessary to write another simple one:
- only deploy the application in one cluster
- emphasise `git`: edit file --> git add --> git commit --> git push  --> (seconds later) a new application!
- just changes some specs for the application: for example, the env filed and verify it's changed.

I think it might be better to teach Kubevela users the basic feature of KubeVela GitOps. And of course, keeping this advanced turotial.

File: docs/case-studies/gitops.md (hunk 3 above)
Comment (FogDong) [reply docs/case-studies/gitops.md]: I will update the kustomize in vela show later, thanks for the report

File: docs/case-studies/gitops.md (hunk 4 above)
Comment (FogDong) [reply docs/case-studies/gitops.md]: The application in the `default` namespace won't be modified or recreated, you can see the age of the apps:

```shell
$ kubectl get application -A
NAMESPACE     NAME                  COMPONENT                               TYPE         PHASE     HEALTHY   STATUS   AGE
default       first-vela-workflow   express-server                          webservice   running   true               10m
default       git-app               gitops                                  kustomize    running   true               10m
prod          first-vela-workflow   express-server                          webservice   running   true               8s
```

Comment (zzxwill) [reply docs/case-studies/gitops.md]: Should the application be deleted? As the user has changed the source code, he/she has changed, or we can say, lost the single-truth-of-source, the application should be deleted.

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
+---
+title:  GitOps 部署
+---
+
+GitOps 作为持续交付的方式之一，使开发人员可以通过更改 Git 仓库中的代码来自动部署应用，本节将介绍如何使用 KubeVela 完成 GitOps 部署。
+
+## 准备工作
+
+首先，准备一个 Git 仓库，里面是需要部署的应用文件，可参考 [示例仓库](https://github.com/FogDong/KubeVela-GitOps-Demo)：
```
Comment (BinaryHB0916) [inline i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]: 这里放到个人 repo 里不合适。可以在 Case Study 目录下新建一个 Demo 目录，然后放 YAML 文件来引用

Comment (BinaryHB0916) [reply i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]: 如果是想体现 GitOps 要基于一个新 repo 这个意思，可以在 oam-dev org 下新建一个 demo repo。cc @wonderflow 也判断一下。

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+接着，在集群中部署一个 secret，里面填写你的 Git 账号名与密码：
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-secret
+type: kubernetes.io/basic-auth
+stringData:
+  username: your username
```
Comment (BinaryHB0916) [inline i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]:
```suggestion
  username: <your username>
```

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  password: your password
```
Comment (BinaryHB0916) [inline i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]:
```suggestion
  password: <your password>
```

Comment (BinaryHB0916) [reply i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]: 为了以示区分，在需要用户自行填写相关数据的字段里，推荐用 <> 括起来

File: docs/case-studies/gitops.md (hunk 4 above)
Comment (FogDong) [reply docs/case-studies/gitops.md]: Rewrite the whole docs, please review the zh version and I'll update en version later.

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+title:  GitOps 部署
+---
+
+GitOps 通过实现持续交付，使开发人员可以通过直接更改 Git 仓库中的代码来自动部署应用，在提高部署生产力的同时也通过分支回滚等能力提高了可靠性。 KubeVela 作为一个应用交付系统，同样可以支持 GitOps 能力，本节将介绍如何使用 KubeVela 完成 GitOps 自动部署。
+
+## 准备代码仓库
+
+首先，准备一个 Git 仓库，里面含有一个 `Application` 配置文件，一些源代码以及对应的 Dockerfile。
+
+代码的实现逻辑非常简单，会启动一个服务，并显示对应的 Version 版本。而在 `Application` 当中，我们会通过一个 `webservice` 类型的 component 启动该服务，并添加一个 `ingress` 的 traits 以方便访问。
```
Comment (BinaryHB0916) [inline i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]:
```suggestion
代码的实现逻辑非常简单，会启动一个服务，并显示对应的 Version 版本。而在 `Application` 当中，我们会通过一个 `webservice` 类型的组件启动该服务，并添加一个 `ingress` 的运维特征以方便访问。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+        port: 8088
+      traits:
+        - type: ingress
+          properties:
+            domain: testsvc.example.com
+            http:
+              /: 8088
+```
+
+我们希望用户改动代码进行提交后，自动构建出镜像并推送到镜像仓库。这一步 CI 可以通过集成 GitHub Actions、Jenkins 或者其他 CI 工具来实现。在本例中，我们通过借助 GitHub Actions 来完成持续集成。具体的代码文件及配置可参考 [示例仓库](https://github.com/oam-dev/kubevela/samples)。
```
Comment (BinaryHB0916) [inline i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]: 示例仓库的 url 不对。

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+kind: Application
+metadata:
+  name: first-vela-workflow
+  namespace: default
+spec:
+  components:
+    - name: test-server
+      type: webservice
+      properties:
+        # replace the imagepolicy `default:gitops` with your policy later
```
Comment (BinaryHB0916) [inline i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]: 英文注释翻译一下

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+> ```shell
+> jobs:
+>  publish:
+>    if: "!contains(github.event.head_commit.message, 'Update image automatically')"
+> ```
+
+重新查看集群中的应用，可以看到经过一段时间后，`Application` 的镜像已经被更新。通过 `curl` 对应的 `Ingress` 查看当前版本：
+
+```shell 
+$ curl -H "Host:testsvc.example.com" http://your-ip
```
Comment (BinaryHB0916) [inline i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]:
```suggestion
$ curl -H "Host:testsvc.example.com" http://<your-ip>
```

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+        # 通过 policy 筛选出最新的镜像 Tag 并用于更新
+        policy:
+          numerical:
+            order: asc
+```
+
+将上述文件部署到集群中后，查看集群中的应用，可以看到，应用 `git-app` 自动拉取了 Git 仓库中的应用配置并部署到了集群中：
+
+```shell
+$ kubectl get application
```
Comment (BinaryHB0916) [inline i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]: 使用 vela ls 命令来查看应用状态，引导用户习惯使用 vela CLI。

我们会逐渐把 vela CLI 功能做强，终态后用户不再需要使用 kubectl。

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+            domain: testsvc.example.com
+            http:
+              /: 8088
+```
+
+我们希望用户改动代码进行提交后，自动构建出镜像并推送到镜像仓库。这一步 CI 可以通过集成 GitHub Actions、Jenkins 或者其他 CI 工具来实现。在本例中，我们通过借助 GitHub Actions 来完成持续集成。具体的代码文件及配置可参考 [示例仓库](https://github.com/oam-dev/kubevela/samples)。
+
+## 配置秘钥信息
+
+在新的镜像推送到镜像仓库后，KubeVela 会识别到新的镜像，并更新仓库及集群中的 `Application` 配置文件。因此，我们需要一个含有 Git 信息的 Secret，使 Vela 向 Git 仓库进行提交。
```
Comment (BinaryHB0916) [inline i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]:
```suggestion
在新的镜像推送到镜像仓库后，KubeVela 会识别到新的镜像，并更新仓库及集群中的 `Application` 配置文件。因此，我们需要一个含有 Git 信息的 Secret，使 KubeVela 向 Git 仓库进行提交。
```

Comment (BinaryHB0916) [reply i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]: 正式文档里全部使用完整正确的产品称谓 KubeVela，而不是 vela 或者 Vela。

下面还有多处也需要修改。

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md (hunk 9 above)
Comment (FogDong) [reply i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]: 还没合进去

File: docs/case-studies/gitops.md [hunk 14]
```diff
@@ -0,0 +1,143 @@
+---
+title:  GitOps with KubeVela
+---
+
+GitOps is a continuous delivery method that allows developers to automatically deploy applications by changing code in a Git repository. In this section, you will learn how to use KubeVela for GitOps deployment.
```
Comment (hongchaodeng) [inline docs/case-studies/gitops.md]:
```suggestion
GitOps is a continuous delivery method that allows developers to automatically deploy applications by changing code in a Git repository. In this section, you will learn how to use KubeVela to do GitOps.
```

File: docs/case-studies/gitops.md [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+
+## Preparation
```
Comment (hongchaodeng) [inline docs/case-studies/gitops.md]:
```suggestion
## Setup
```

File: docs/case-studies/gitops.md [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+
+First, prepare a Git Repository with `Application` files that need to be applied, please refer to [KubeVela-GitOps-Demo](https://github.com/FogDong/KubeVela-GitOps-Demo):
```
Comment (hongchaodeng) [inline docs/case-studies/gitops.md]:
```suggestion
First, setup a Git Repository with `Application` files that need to be applied, please refer to [KubeVela-GitOps-Demo](https://github.com/FogDong/KubeVela-GitOps-Demo):
```

File: docs/case-studies/gitops.md (hunk 1 above)
Comment (hongchaodeng) [inline docs/case-studies/gitops.md]:
```suggestion
Then create a secret with your Git username and password:
```

File: docs/case-studies/gitops.md [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+kind: Secret
+metadata:
+  name: my-secret
+type: kubernetes.io/basic-auth
+stringData:
+  username: your username
+  password: your password
+```
+
+## Write the Application that can apply automatically
```
Comment (hongchaodeng) [inline docs/case-studies/gitops.md]:
```suggestion
## Apply the Application that can sync with your Git
```

File: docs/case-studies/gitops.md [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+
+```shell
+$ kubectl get application
+
+NAME                  COMPONENT        TYPE         PHASE     HEALTHY   STATUS   AGE
+first-vela-workflow   express-server   webservice   running   true               1s
+git-app               gitops           kustomize    running   true               3s
+```
+
+## Edit the configuration
```
Comment (hongchaodeng) [inline docs/case-studies/gitops.md]:
```suggestion
## Update Application to deploy to more clusters
```

**Review by hongchaodeng — APPROVED** [review]
这个 GitOps 的文档写得简洁扼要，建议将它写成 Best practices 发表！

File: docs/case-studies/gitops.md [hunk 19]
```diff
@@ -0,0 +1,154 @@
...
+kind: Secret
+metadata:
+  name: my-secret
+type: kubernetes.io/basic-auth
+stringData:
+  username: <your username>
+  password: <your password>
+```
+
+## Write the automatic apply file
```
Comment (hongchaodeng) [inline docs/case-studies/gitops.md]: Create the Application that sync with Git

File: docs/case-studies/gitops.md [hunk 20]
```diff
@@ -0,0 +1,154 @@
...
+
+After completing the basic configuration above, we can create a new automatic apply file locally and associate the corresponding Git repository and image registry information:
```
Comment (hongchaodeng) [inline docs/case-studies/gitops.md]:
```suggestion
After completing the basic configuration above, we can create an Application file that syncs with the corresponding Git repository and image registry information:
```

File: docs/case-studies/gitops.md [hunk 21]
```diff
@@ -0,0 +1,154 @@
...
+```
+
+We can `curl` the `Ingress` to see the current version in code:
+
+```shell 
+$ curl -H "Host:testsvc.example.com" http://<your-ip>
+Version: 0.1.5
+```
+
+## Modify the code
```
Comment (hongchaodeng) [inline docs/case-studies/gitops.md]:
```suggestion
## Modify the code to trigger automatic deployment
```

File: docs/case-studies/gitops.md [hunk 22]
```diff
@@ -0,0 +1,154 @@
...
+Re-check the `Application` in cluster, we can see that the image of the `Application` has been updated after a while. We can `curl` to `Ingress` to see the current version:
+
+```shell 
+$ curl -H "Host:testsvc.example.com" http://<your-ip>
+Version: 0.1.6
+```
+
+The `Version` has been updated successfully! Now we're done with everything from changing the code to automatically applying to the cluster.
+
+KubeVela gets the latest information from the code repository and the image repository at regular intervals, depending on the interval you specify:
```
Comment (hongchaodeng) [inline docs/case-studies/gitops.md]:
```suggestion
KubeVela polls the latest information from the code and image repo periodically (at an interval that can be customized):
```

File: i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md (hunk 9 above)
Comment (wonderflow) [reply i18n/zh/docusaurus-plugin-content-docs/current/case-studies/gitops.md]: 用 oam-dev/samples ，不要跟KubeVela主仓库混在一起

File: docs/case-studies/gitops.md [hunk 23]
```diff
@@ -0,0 +1,154 @@
...
+        port: 8088
+      traits:
+        - type: ingress
+          properties:
+            domain: testsvc.example.com
+            http:
+              /: 8088
+```
+
+We want users to build the image and push it to the image registry after changing the code, so we need to integrate with a CI tool like GitHub Actions or Jenkins to do it. In this example, we use GitHub Actions to build the image. For the code and configuration file, please refer to [Example Repo](https://github.com/oam-dev/kubevela/samples).
```
Comment (wonderflow) [inline docs/case-studies/gitops.md]: 代码仓库改一下

**Review by wonderflow — APPROVED** [review]

**Pull request merged**